src/Widgets/AppEntry.vala

```diff
@@ -101,7 +110,11 @@ public class Notifications.AppEntry : Gtk.ListBoxRow {
         });
 
         clear_btn_entry.clicked.connect (() => {
-            clear (); // Causes notification list to destroy this app entry after clearing its notification entries
+            clear_btn_image.get_style_context ().add_class ("active");
+            GLib.Timeout.add (550, () => {
```

Comment:
We should run `clear_all_notification_entries ();` before the timeout so that the response is still immediate and we're only waiting for the animation to finish to clear the header
```suggestion
            clear_all_notification_entries ();
            GLib.Timeout.add (600, () => {
```